### PR TITLE
Support all the extension functions in the python client

### PR DESCRIPTION
**Pull request by tavallaie**
## Changes Made

- Added `drop_queue`, `purge_queue`, `detach_archive`, and `validate_queue_name`.
- Added `send_batch`, `read_batch`, `read_with_poll`, `archive_batch`, `delete_batch`, and `set_vt`.
- Added `list_queues`.
- Updated Docker image in Makefile.

## Related Issue
- Addresses #216.

### CODE REVIEW

File: tembo-pgmq-python/Makefile [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+
+check-poetry-version:
+	@poetry_version=$$(poetry --version | awk '{print $$3}'); \
+	required_version=$(MIN_POETRY_VERSION); \
+	if [ $$(printf '%s\n' "$$required_version" "$$poetry_version" | sort -V | head -n1) != "$$required_version" ]; then \
+		echo "Poetry version $$poetry_version is not sufficient. Please install Poetry version >= $(MIN_POETRY_VERSION)."; \
+		exit 1; \
+	fi
+
+format: check-poetry-version
```
Comment (ChuckHend) [inline tembo-pgmq-python/Makefile:14]: Thoughts on managing the project with https://asdf-vm.com/? Not in the PR, but in the future. I find it very useful for managing the various toolchain versions, like python and poetry.

File: tembo-pgmq-python/README.md [hunk 2]
```diff
@@ -121,8 +159,39 @@ print(popped_message)
...
+```python
+dropped: bool = queue.drop_queue("my_queue")
+print(f"Queue dropped: {dropped}")
+```
+
+### Validate the length of a queue name
+
+```python
+queue.validate_queue_name("my_queue")
+```
```
Comment (v0idpwn) [inline tembo-pgmq-python/README.md:180]: This one doesn't need to be exposed by the python library, I think? WDYT, @ChuckHend? I can see it being useful for "dynamic names" but idk

**Review by v0idpwn — APPROVED** [review]

**Review by ChuckHend — APPROVED** [review]

File: tembo-pgmq-python/README.md (hunk 2 above)
Comment (ChuckHend) [reply tembo-pgmq-python/README.md:180]: It is not an essential in the clients but I could see some people finding it useful.

File: tembo-pgmq-python/Makefile (hunk 1 above)
Comment (tavallaie) [reply tembo-pgmq-python/Makefile:14]: did you use it with poetry directly? 
I update my poetry and add version control for sake of lower version, but if it has good impact on developing experience I will be glad to use it.

**Comment by v0idpwn** [comment]
Thanks @tavallaie!

File: tembo-pgmq-python/Makefile (hunk 1 above)
Comment (ChuckHend) [reply tembo-pgmq-python/Makefile:14]: I install python and poetry using asdf, like this.

```
asdf install poetry 1.7.1
asdf install python 3.11.0
```

Then, specify the versions in the `.tool-versions` file. 

```text
python 3.11.0
poetry 1.7.1
```

`.tool-versions` can be committed to source control, but it is currently not for this project. I just found it useful for managing python virtual environments.

**Pull request merged**